# <span style ="font-family:Gill Sans"><font color='orange'>INSTAGRAM LIKES REPORT</font></span>

<span style ="font-family:Georgia">**Andrew Gudz**</span>

<span style ="font-family:Georgia">09/20/22</span>

<span style ="font-family:Georgia">Technology has greatly evolved to become a tool to collect data. Through trackers and bots, companies can collect information on a user's activity on any device. With this collected data, companies can advertise to the user, like posting advertisements related to topics of interest. Like many companies, Instagram collects data from user activity. By examining the "like activity" of Dr. Silva, a reader can understand how Instagram collects data and how reliable the bots are.</span>

<span style ="font-family:Georgia">Before seeing how many likes are recorded, its important to see the plain data. This is accomplished by opening the related json file titled "likes.json". The first task is to create a few variables on how the json file and the panda system will activate in the Python system. Pandas are a system of code that will greatly help sort the data later in the report.

In [6]:
import pandas as pd
import json

<span style ="font-family:Georgia">Once those variables are setup correctly, two more variables are created to open the json. By establishing the variable "j" as the "open command" there is now a feasible way to open the file. And the "likes_dat" variable acts as a vessel so the data can finally be open.</span>

In [7]:
with open ("/Users/andrewgudz/Data_Fall_2022/technogecko_20200714_toshare/likes.json") as j:
    likes_dat = json.load(j)

<span style ="font-family:Georgia">And eureka! The data is now viewable.</span>

In [8]:
likes_dat

{'media_likes': [['2020-07-11T04:39:28+00:00', 'ball_doesnt_lie'],
  ['2020-07-11T04:39:05+00:00', 'ball_doesnt_lie'],
  ['2020-07-05T17:25:44+00:00', 'ali_saurusrex'],
  ['2020-07-03T03:40:02+00:00', 'cacandassociates'],
  ['2020-06-25T17:41:50+00:00', 'cacandassociates'],
  ['2020-06-22T23:01:55+00:00', 'reams_esq'],
  ['2020-06-08T15:05:46+00:00', 'emmyr0o'],
  ['2020-06-07T12:46:29+00:00', 'ali_saurusrex'],
  ['2020-06-02T01:03:28+00:00', 'colin_storm'],
  ['2020-05-25T16:38:14+00:00', 'ali_saurusrex'],
  ['2020-05-19T23:38:40+00:00', 'colin_storm'],
  ['2020-05-18T13:42:30+00:00', 'emmyr0o'],
  ['2020-05-14T13:51:03+00:00', 'emmyr0o'],
  ['2020-05-12T21:31:12+00:00', 'cacandassociates'],
  ['2020-05-11T05:07:31+00:00', 'inalull'],
  ['2020-05-07T18:07:52+00:00', 'reams_esq'],
  ['2020-05-06T00:33:58+00:00', 'inalull'],
  ['2020-04-30T19:54:48+00:00', 'emmyr0o'],
  ['2020-04-28T12:58:14+00:00', 'inalull'],
  ['2020-04-28T03:28:35+00:00', 'cacandassociates'],
  ['2020-04-26T06:41:17

<span style ="font-family:Georgia">Can the data now tell the reader how many likes Dr. Silva has given? Well...not exactly. The data would be easier to sort and read if it was sorted in a table. This is accomplished by creating a panda "DataFrame". For practice, the "Timestamp" section of the data was put through a "DataFrame". The "Timestamp" section acts as a header for the soon to be table along with specific data collected from the previous variable "likes_dat". The "['media_likes']" further specifies the Timestamp data from this table.<span>

In [9]:
likes_dat_df = pd.DataFrame({'Timestamp': likes_dat['media_likes']})

<span style ="font-family:Georgia">And once a new variable is created with the "DataFrame" command the table is now viewable.<span>

In [10]:
likes_dat_df

Timestamp
0     [2020-07-11T04:39:28+00:00, ball_doesnt_lie]
1     [2020-07-11T04:39:05+00:00, ball_doesnt_lie]
2       [2020-07-05T17:25:44+00:00, ali_saurusrex]
3    [2020-07-03T03:40:02+00:00, cacandassociates]
4    [2020-06-25T17:41:50+00:00, cacandassociates]
..                                             ...
330            [2013-02-05T02:58:46+00:00, natgeo]
331       [2013-02-05T02:22:24+00:00, aroseroar16]
332       [2013-02-05T01:29:31+00:00, aroseroar16]
333     [2013-02-04T17:42:04+00:00, ali_saurusrex]
334      [2013-02-04T17:39:15+00:00, a_matt_silva]

[335 rows x 1 columns]

<span style ="font-family:Georgia">Can a table based on how many likes now be created? Not yet. Its best to understand what categories of data are in the json. The first category, which was just created in a table, is the "Timestamps"; essential when the like happened. The second category is the "Accountnames"; the user name of each liked post. These two categories can be separated to better create the table.<span>

<span style ="font-family:Georgia">By creating two new variables, "timestamp" and "account_name", the data can be separated and used specifically for columns for the final table. Similar to a for loop in javascript the variable "x" can specify that the data should be taken from the previous table "likes_dat_df" and the specific data of ['Timestamp'].<span>

In [11]:
timestamp = [x[0] for x in likes_dat_df['Timestamp']]

<span style ="font-family:Georgia">Now all the "Timestamps" can be displayed through a variable.</span>

In [12]:
timestamp

['2020-07-11T04:39:28+00:00',
 '2020-07-11T04:39:05+00:00',
 '2020-07-05T17:25:44+00:00',
 '2020-07-03T03:40:02+00:00',
 '2020-06-25T17:41:50+00:00',
 '2020-06-22T23:01:55+00:00',
 '2020-06-08T15:05:46+00:00',
 '2020-06-07T12:46:29+00:00',
 '2020-06-02T01:03:28+00:00',
 '2020-05-25T16:38:14+00:00',
 '2020-05-19T23:38:40+00:00',
 '2020-05-18T13:42:30+00:00',
 '2020-05-14T13:51:03+00:00',
 '2020-05-12T21:31:12+00:00',
 '2020-05-11T05:07:31+00:00',
 '2020-05-07T18:07:52+00:00',
 '2020-05-06T00:33:58+00:00',
 '2020-04-30T19:54:48+00:00',
 '2020-04-28T12:58:14+00:00',
 '2020-04-28T03:28:35+00:00',
 '2020-04-26T06:41:17+00:00',
 '2020-04-16T22:09:55+00:00',
 '2020-04-15T18:44:15+00:00',
 '2020-04-10T04:46:19+00:00',
 '2020-04-08T23:18:20+00:00',
 '2020-03-29T19:10:28+00:00',
 '2020-03-27T17:31:39+00:00',
 '2020-03-26T21:13:08+00:00',
 '2020-03-23T21:36:47+00:00',
 '2020-03-09T12:01:22+00:00',
 '2020-03-07T22:39:01+00:00',
 '2020-03-06T12:05:32+00:00',
 '2020-02-08T03:57:54+00:00',
 '2020-02-

<span style ="font-family:Georgia">Similar to the "timestamp" variable, an "account_name" variable is created and all that specific data can now be viewed.</span>

In [13]:
account_name = [x[1] for x in likes_dat_df['Timestamp']]

In [14]:
account_name

['ball_doesnt_lie',
 'ball_doesnt_lie',
 'ali_saurusrex',
 'cacandassociates',
 'cacandassociates',
 'reams_esq',
 'emmyr0o',
 'ali_saurusrex',
 'colin_storm',
 'ali_saurusrex',
 'colin_storm',
 'emmyr0o',
 'emmyr0o',
 'cacandassociates',
 'inalull',
 'reams_esq',
 'inalull',
 'emmyr0o',
 'inalull',
 'cacandassociates',
 'inalull',
 'jennjohnn',
 'cacandassociates',
 'ali_saurusrex',
 'cacandassociates',
 'cacandassociates',
 'cacandassociates',
 'inalull',
 'cacandassociates',
 'ali_saurusrex',
 'inalull',
 'cacandassociates',
 'poorlydrawnlines',
 'athenalemon',
 'inalull',
 'orangekoala2',
 'cacandassociates',
 'mycaninelife',
 'ali_saurusrex',
 'reams_esq',
 'cacandassociates',
 'inalull',
 'ali_saurusrex',
 'emmyr0o',
 'ali_saurusrex',
 'inalull',
 'reams_esq',
 'leleebabe',
 'inalull',
 'ali_saurusrex',
 'colin_storm',
 'colin_storm',
 'cacandassociates',
 'cacandassociates',
 'ali_saurusrex',
 'silva_c097',
 'silva_c097',
 'leleebabe',
 'leleebabe',
 'silva_c097',
 'mtb_crohnert

<span style ="font-family:Georgia">Now that the two variables are created, a categorized table can now be formed. "timestamp" and "account_name" variables can be used as columns for a new "DataFrame" chart.</span>

In [15]:
table_df = pd.DataFrame({'Timestamps':timestamp,'Accountnames':account_name})

<span style ="font-family:Georgia">And now finally the data is now in a readable chart.</span>

In [16]:
table_df

Timestamps      Accountnames
0    2020-07-11T04:39:28+00:00   ball_doesnt_lie
1    2020-07-11T04:39:05+00:00   ball_doesnt_lie
2    2020-07-05T17:25:44+00:00     ali_saurusrex
3    2020-07-03T03:40:02+00:00  cacandassociates
4    2020-06-25T17:41:50+00:00  cacandassociates
..                         ...               ...
330  2013-02-05T02:58:46+00:00            natgeo
331  2013-02-05T02:22:24+00:00       aroseroar16
332  2013-02-05T01:29:31+00:00       aroseroar16
333  2013-02-04T17:42:04+00:00     ali_saurusrex
334  2013-02-04T17:39:15+00:00      a_matt_silva

[335 rows x 2 columns]

<span style ="font-family:Georgia">But how many likes were there total? There is no category of data called "likes" so how can the likes be viewed? The answer is revealed by counting the amount of "Timestamps". If "Accountnames" were counted there could be repeated names, meaning that some likes will not be counted. By using the "groupby" command and its further actions of "count()" and "sum()" the Timestamps can be counted. Through the amount of Timestamps there are a total of 335 likes.</span>

In [17]:
table_df.groupby('Timestamps').count().sum()

Accountnames    335
dtype: int64

<span style ="font-family:Georgia">So the answer to how many likes was found, but are the findings reasonable? Not quite. As a surface-level issue, the json file only contains likes from 2013 to summer of 2020. Meaning that any likes from the rest of 2020 and onwards are not recorded. This leads to a greater issue of the Instagram trackers; are the trackers biased on collecting certain categories of data. Because trackers and bots were programed by humans, they have the ability to target certain sets of activity and potentially advertise other posts. For example, from the likes data, Dr. Silva liked quite a few posts from user "aroseroar16". Could this mean other liked posts from other "Accountnames" be related to liked posts from "aroseroar16", such as similar content or if they are related to Dr. Silva in anyway? And are liked posts based on least liked categories are unaccounted? Unfortunately, the original json file does not include a category of data based on content of each liked post, a fault of the findings. While there is a solid number of how many liked posts there are, some might be unaccounted for due to biases of certain trackers.</span>